rhino/src/main/java/org/mozilla/javascript/ES6Generator.java

```diff
@@ -199,7 +199,8 @@ private Scriptable resumeDelegeeReturn(Context cx, Scriptable scope, Object valu
         try {
             // Call "return" but don't throw if it can't be found
             Object retResult = callReturnOptionally(cx, scope, value);
-            // Per spec, treat undefined return value same as null
+            // Treat undefined return value same as null
+            // See https://tc39.es/ecma262/#sec-iteratorclose (7.4.11, 4b)
```

Comment:
That is a somewhat confusing spec reference to give as the text of 4b is:
> b. If return is undefined, return ? completion.

The conversion of `null` to `undefined` is occurring only because 7.4.11.3 specifies that the abstract operation `GetMethod` is used, and *that* specifies that `undefined` is returned is the method is `null` or `undefined`. I think it's a good idea to join the dots here.

Reply:
When referencing the EcmaScript spec, reference the specific version of the spec you're working off of, because chapter and section numbers within the spec can change from version to version